### PR TITLE
Add custom pager handling

### PR DESCRIPTION
**Pull request by GokceGK**
Related Github issue https://github.com/stackitcloud/stackit-cli/issues/154

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,26 @@ stackit config list
...
 
+## Customization
+
+### Using **less** as a pager
+
+STACKIT CLI uses the `less` as default pager. 
+When using `less` as a pager, STACKIT CLI will automatically pass following options
+- -F, --quit-if-one-screen - Less will automatically exit if the entire file can be displayed on the first screen.
+- -S, --chop-long-lines - Lines longer than the screen width will be chopped rather than being folded.
+- -w, --hilite-unread - Temporarily highlights the first "new" line after a forward movement of a full page.
```
Comment (vicentepinto98) [inline README.md:140]: It's missing the newly introduced argument "-R" for the colors

File: README.md [hunk 2]
```diff
@@ -129,6 +129,26 @@ stackit config list
 
 You can also edit the configuration file manually.
 
+## Customization
```
Comment (vicentepinto98) [inline README.md:132]: Re-structure this section to be more clear like this:
```
## Customization

### Pager

To specify a custom pager, use the `PAGER` environment variable.

If the variable is not set, STACKIT CLI uses the `less` as default pager.
When using `less` as a pager .....

[IMPORTANT] 
....
```

**Comment by vicentepinto98** [comment]
Please link the GH issue that this PR solves

File: internal/pkg/print/print.go [hunk 3]
```diff
@@ -165,9 +165,15 @@ func (p *Printer) PagerDisplay(content string) error {
 	}
 	pagerCmd := exec.Command("less", "-F", "-S", "-w")
 
+	pager, pagerExists := os.LookupEnv("PAGER")
```
Comment (vicentepinto98) [inline internal/pkg/print/print.go:168]: It this a commonly known env var?

File: README.md [hunk 4]
```diff
@@ -129,6 +129,26 @@ stackit config list
...
+STACKIT CLI uses the `less` as default pager. 
+When using `less` as a pager, STACKIT CLI will automatically pass following options
+- -F, --quit-if-one-screen - Less will automatically exit if the entire file can be displayed on the first screen.
+- -S, --chop-long-lines - Lines longer than the screen width will be chopped rather than being folded.
+- -w, --hilite-unread - Temporarily highlights the first "new" line after a forward movement of a full page.
+
+
+>[IMPORTANT]
+> These options will not be added if:
+> - The pager is not named `less`.
```
Comment (joaopalet) [inline README.md:145]: Can't the user pass in flags for their custom pager by setting `PAGER` to something like `less -x`?

Comment (joaopalet) [inline README.md:145]: Can't the user pass in flags for their custom pager by setting `PAGER` to something like `less -x`?

Comment (joaopalet) [reply README.md:145]:
```suggestion
> These options will not be added automatically if a custom pager is defined.
```

Comment (GokceGK) [reply README.md]: In the current state of implementation, no args can be passed. Should this be possible? 

If we want to support args, we need to split the env variable content as command and args.

File: internal/pkg/print/print.go (hunk 3 above)
Comment (GokceGK) [reply internal/pkg/print/print.go:168]: yes it is, a lot of tools uses PAGER env variable.

File: README.md (hunk 1 above)
Comment (GokceGK) [reply README.md:144]: Added a line to explain the -R option.

File: README.md (hunk 2 above)
Comment (GokceGK) [reply README.md:133]: Updated the section structure as suggestion

File: README.md (hunk 4 above)
Comment (joaopalet) [reply README.md]: Even if the env variable is defined as `PAGER="less -x -y -z"`?

Comment (GokceGK) [reply README.md]: yes, even if we pass args like that

Comment (vicentepinto98) [reply README.md]: That's weird :thinking:

Comment (vicentepinto98) [reply README.md]: I would say it's not a must have, but also the other CLIs don't support this?

Comment (GokceGK) [reply README.md]: @vicentepinto98 the one I checked handles the given string differently to detect parameters and pass as args in the exec.command

Comment (GokceGK) [reply README.md]: @joaopalet extended README for the env variables to define parameters as discussed tomorrow :)

File: README.md [hunk 5]
```diff
@@ -130,6 +130,31 @@ stackit config list
...
+
+When using `less` as a pager, STACKIT CLI will automatically pass following options
+- -F, --quit-if-one-screen - Less will automatically exit if the entire file can be displayed on the first screen.
+- -S, --chop-long-lines - Lines longer than the screen width will be chopped rather than being folded.
+- -w, --hilite-unread - Temporarily highlights the first "new" line after a forward movement of a full page.
+- -R, --RAW-CONTROL-CHARS - ANSI color and style sequences will be interpreted.
+
+> These options will not be added automatically if a custom pager is defined.
+> 
+> In that case, users can define the parameters by using the specific environment variable required by the PAGER (if supported by pager).
```
Comment (vicentepinto98) [inline README.md:149]:
```suggestion
> In that case, users can define the parameters by using the specific environment variable required by the `PAGER` (if supported).

```

File: README.md [hunk 6]
```diff
@@ -130,6 +130,31 @@ stackit config list
...
+> 
+> For example, if user sets the PAGER environment variable to less and would like to pass some arguments, LESS environment variable must be used as following.
```
Comment (vicentepinto98) [inline README.md:151]:
```suggestion
> For example, if user sets the `PAGER` environment variable to `less` and would like to pass some arguments, `LESS` environment variable must be used as following:

```

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**